liquibase-core/src/main/java/liquibase/structure/DatabaseObject.java

```diff
@@ -92,5 +92,17 @@ public interface DatabaseObject extends Comparable, LiquibaseSerializable {
      */
     DatabaseObject setAttribute(String attribute, Object value);
 
+    /**
+     * Given a database object, build its fully qualified name by looking up the at its containing objects.
+     * @return the fully qualified name
+     */
+    default String buildFullyQualifiedName() {
```

Comment:
This is probably better named `getFullyQualifiedName` ?

Reply:
I'm also not sure this method should exist. The object model is database agnostic, but the qualified name is often times going to vary depending on the database, and having this simply return a dot-separated string will cause problems as people try to use it in SQL statements or something and it starts failing when the object names need to be quoted etc.

Maybe this should be moved to be a method on Database? And maybe the return type should be a String[] so needed logic like quoting etc. can be applied as needed? Or maybe even there should be a FullyQualifiedName return type so callers can understand what those types are since quoting etc. rules will depend on that? And the generic "just dot them together" could be the toString() method on FullyQualifiedName?

Reply:
Returning a `FullyQualifiedName` object that can be toStringed like you're describing. Is that your preferred solution to this, rather than returning a String[]?